airflow-core/docs/security/security_model.rst

```diff
@@ -115,16 +115,25 @@ Connection configuration users
 ..............................
 
 They configure connections and potentially execute code on workers during DAG execution. Trust is
-required to prevent misuse of these privileges. They have full access
-to sensitive credentials stored in connections and can modify them.
-Access to sensitive information through connection configuration
+required to prevent misuse of these privileges. They have full write-only access
+to sensitive credentials stored in connections and can modify them, but cannot view them.
+Access to write sensitive information through connection configuration
 should be trusted not to be abused. They also have the ability to configure connections wrongly
 that might create a API Server Denial of Service situations and specify insecure connection options
 which might create situations where executing dags will lead to arbitrary Remote Code Execution
 for some providers - either community released or custom ones.
 
 Those users should be highly trusted not to misuse this capability.
 
+.. note::
+
+   Before Airflow 3.0.4, the **Connection configuration users** role had also access to view the sensitive information this has
+   been changed in Airflow 3.0.4 to improve security of the accidental spilling of credentials of the connection configuration
```

Comment:
That's not super accurate, it's really just `3.0.3` that introduced a mistake but that was never the intend for AF3 and maybe we shouldn't highlight too much on that. In 3.0.0, 3.0.1, 3.0.2, the API redacts every field, and nothing is revealed. In AF2 though your statement is correct I believe.

I think we can just compare AF3 and AF2.

Reply:
> I guess we should then update the model to indicate difference between 3 and 2 and yank 3.0.3 eventualy as it does not fulfill the - retroactively updated - model.

I agree with this.